### PR TITLE
[llvm] Update to version 10.0.0 and add new features

### LINKED ISSUE #3377: separate LLVM and Clang ports
The overhead of building clang is way more than building simple LLVM. It does not make sense at all to have clang as part of LLVM. At least there should be an option to tell VCPKG to only build LLVM.  

When you have a port named LLVM it should be LLVM, not LLVM/Clang. You can have another package named Clang or you can add config to LLVM package to compile it with Clang.
Whose idea was that? Seriously?

### LINKED ISSUE #4277: [LLVM] misconfiguration
When using the `llvm-config.exe`, it reports bogus directories, hence projects depending on LLVM will fail to configure. I'm trying to build the [POCL](https://github.com/pocl/pocl) library using the LLVM Vcpkg, but the tools report directories like:

```PowerShell
PS C:\Users\mnagy> .\Source\Repos\vcpkg\installed\x64-windows\tools\llvm\llvm-config.exe --includedir
C:\Users\mnagy\Source\Repos\vcpkg\installed\x64-windows\tools/include
PS C:\Users\mnagy> .\Source\Repos\vcpkg\installed\x64-windows\tools\llvm\llvm-config.exe --libdir
C:\Users\mnagy\Source\Repos\vcpkg\installed\x64-windows\tools\lib
PS C:\Users\mnagy> .\Source\Repos\vcpkg\installed\x64-windows\tools\llvm\llvm-config.exe --bindir
C:\Users\mnagy\Source\Repos\vcpkg\installed\x64-windows\tools\tools\llvm
```
None of these directories exist.

### LINKED ISSUE #4461: Package Request: lld
LLVM Linker: https://github.com/llvm-mirror/lld

### LINKED ISSUE #5153: [llvm] CLANG_INSTALL_PREFIX has wrong value
When doing a `find_package(Clang CONFIG REQUIRED)` in a project using vcpkg's toolchain file, I noticed that Clang's CMake setup is a bit messed up. The specific error is:

```
CMake Error at /build/vcpkg/installed/x64-osx/share/clang/ClangConfig.cmake:18 (include):
  include could not find load file:

    /build/vcpkg/installed/lib/cmake/clang/ClangTargets.cmake
Call Stack (most recent call first):
  /build/vcpkg/scripts/buildsystems/vcpkg.cmake:247 (_find_package)
  CMakeLists.txt:78 (find_package)
```

In `ClangConfig.cmake` on we have:

```cmake
set(CLANG_CMAKE_DIR "${CLANG_INSTALL_PREFIX}/lib/cmake/clang")
...
include("${CLANG_CMAKE_DIR}/ClangTargets.cmake")
```

This leads me to believe that `CLANG_INSTALL_PREFIX` is incorrect. Specifically, the installation prefix lacks the os/target `x64-osx`, and perhaps other things. This seems to be the same problem as what is listed here: https://answers.launchpad.net/ubuntu/+source/llvm-toolchain-5.0/+question/663810

### LINKED ISSUE #8347: [llvm] Enable rtti to solve linker error
****

**Is your feature request related to a problem? Please describe.**
When using llvm as a library, an application using llvm fails with a linker error, specifically:
`CMakeFiles/SPIRVToLLVM.dir/SpirvJit.cpp.o:(.data.rel.ro._ZTIN4llvm9ErrorInfoINS_9ErrorListENS_13ErrorInfoBaseEEE[_ZTIN4llvm9ErrorInfoINS_9ErrorListENS_13ErrorInfoBaseEEE]+0x10): undefined reference to typeinfo for llvm::ErrorInfoBase`

**Proposed solution**
Add `-DLLVM_ENABLE_RTTI=1` to the portfile.

### LINKED ISSUE #8463: [llvm] update to 9.0.0
Library name: LLVM

New version number: 9.0.0

Please update to the latest stable version of LLVM!

### LINKED ISSUE #9670: [llvm] build failure when installed alongside iconv, macOS
hopefully I'll be fixing this soon, as well as maybe updating to 9.0

might be fixed by #8521

### PR DESCRIPTION
**Pull request by yurybura**
LLVM 10 release is coming soon.
So, this PR updates LLVM to 10.0.0 version.
Other changes:
- use Github monorepo;
- support sub-projects compiler-rt, clang-tools-extra, lld, openmp, polly;
- exclude tests and examples from build to reduce disk space usage;
- enable RTTI by default

Related to #8521

Fixes #3377
Fixes #4277
Fixes #4461
Fixes #5153
Fixes #8347
Fixes #8463
Fixes #9670

### CODE REVIEW

**Comment by yurybura** [comment]
Dear @PhoebeHui, could you help to solve Windows build-bot issue?
Linker generates out of memory errors:

LINK : fatal error LNK1201: error writing to program database 'C:\vsts\_work\4\s\buildtrees\llvm\x64-windows-dbg\bin\llvm-dwp.pdb'; check for insufficient disk space, invalid path, or insufficient privilege

LINK : fatal error LNK1102: out of memory

LLVM is really big project. I can disable some features by default, but it isn't right way I think.

Thank you!

**Comment by PhoebeHui** [comment]
@strega-nil, could you help take a look at above issue?

**Comment by PhoebeHui** [comment]
@yurybura, thanks for your contribution!

Noted that the following triplets disabled on https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt#L1002, if you fix the issue with specific triplet, please remove it.

llvm:arm64-windows=fail
llvm:arm-uwp=fail
llvm:x64-uwp=fail
llvm:x64-linux=ignore
# installing iconv makes building llvm fail; needs to be fixed
llvm:x64-osx=ignore

**Comment by yurybura** [comment]
@PhoebeHui, let me try to fix failed CI checks.
Thank you!

**Comment by yurybura** [comment]
@PhoebeHui, I've already tested my changes for x64-windows and x64-linux triplets on my local machine.

Unfortunately I cannot test with CI bot...
After enable x64-linux CI, the build fails with the same problem (low amount of memory for link):
```
collect2: fatal error: ld terminated with signal 9 [Killed]
compilation terminated. 
```

**Comment by PhoebeHui** [comment]
llvm is a complicated and big project, it brings a stress to CI test machines, @strega-nil, any suggestions about the issue?

**Comment by JackBoosY** [comment]
This PR is awesome, thanks for this PR!

**Comment by strega-nil** [comment]
@PhoebeHui huh! Well, that's unfortunate. Lemme think about this!

File: ports/llvm/portfile.cmake [hunk 1]
```diff
@@ -1,132 +1,130 @@
...
-    SHA512 1602343b451b964f5d8c2d6b0654d89384c80d45883498c5f0e2f4196168dd4a1ed2a4dadb752076020243df42ffe46cb31d82ffc145d8e5874163cbb9686a1f
-)
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 
-vcpkg_extract_source_archive_ex(
+vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
-    ARCHIVE ${ARCHIVE}
+    REPO llvm/llvm-project
+    REF llvmorg-10.0.0-rc3
```
Comment (BillyONeal) [inline ports/llvm/portfile.cmake]: Is this really updating to an RC version rather than a release version?

Comment (yurybura) [reply ports/llvm/portfile.cmake]: @BillyONeal
Yes, but I hope it will be fixed soon (now RC4 is available). I wanted to start review as soon as possible because it may take long time (~weeks). Can you have a look on other changes?

Comment (BillyONeal) [reply ports/llvm/portfile.cmake]: My cmake-fu isn't good enough for me to feel confident reviewing this; I'm new on this project and wouldn't want to say much about it. @strega-nil just asked me to review this and I pointed out the only thing that looked strange to me. :)

File: ports/llvm/portfile.cmake [hunk 2]
```diff
@@ -1,132 +1,130 @@
...
-if (EXISTS ${CURRENT_PACKAGES_DIR}/share/llvm/LLVMConfig.cmake)
-    file(READ ${CURRENT_PACKAGES_DIR}/share/llvm/LLVMConfig.cmake LLVM_TOOLS_MODULE)
-    string(REPLACE "\${LLVM_INSTALL_PREFIX}/bin" "\${LLVM_INSTALL_PREFIX}/tools/llvm" LLVM_TOOLS_MODULE "${LLVM_TOOLS_MODULE}")
-    file(WRITE ${CURRENT_PACKAGES_DIR}/share/llvm/LLVMConfig.cmake "${LLVM_TOOLS_MODULE}")
-endif()
+# All LLVM tools should be located in the bin folder because llvm-config expects to be inside a bin dir.
+# So, rename subfolder `/tools/${PORT}` to `/bin` back because there is no way to avoid this in vcpkg_fixup_cmake_targets.
+foreach(_target IN LISTS _release_targets _debug_targets)
+    file(READ ${_target} _contents)
+    string(REPLACE "{_IMPORT_PREFIX}/tools/${PORT}" "{_IMPORT_PREFIX}/bin" _contents "${_contents}")
```
Comment (d-winsor) [inline ports/llvm/portfile.cmake]: I am not familiar with vcpkg but I enlisted in the PR to try out llvm.10 and ran into an issue in Debug. The right files exist in **vcpkg\installed\x64-windows\bin** for release but there are none in **vcpkg\installed\x64-windows\debug\bin** for debug. My guess is that is happening here?

Comment (d-winsor) [reply ports/llvm/portfile.cmake]: I imagine this line should be different for debug as can be seen in the old changes on line 98. Maybe something like:

```cmake
set(_bin_prefix)
if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
    set(_bin_prefix bin)
...
if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
    set(_bin_prefix debug/bin)
...
string(REPLACE "{_IMPORT_PREFIX}/tools/${PORT}" "{_IMPORT_PREFIX}/${_bin_prefix}" _contents "${_contents}")
```

Comment (yurybura) [reply ports/llvm/portfile.cmake]: @d-winsor, Thank you for comments!
This version of port (like the old one) doesn't install binaries for debug configuration. Debug binaries are removed at line 118. Do you really need debug version of tools? Let me know how are you going to use them.
Thank you.

Comment (d-winsor) [reply ports/llvm/portfile.cmake]: The bin needs to be installed for the debug version however it is no longer being copied to `${CURRENT_PACKAGES_DIR}/debug/tools/llvm` and have the cmake files updated to reference this path. Instead they remain in the bin are deleted in the REMOVE_RECURSE call below.

The absence of these binaries causes the find_package to fail because it can't find the *.dlls. I was able to resolve this by removing the `${CURRENT_PACKAGES_DIR}/debug/bin` from the REMOVE_RECURSE and adding the line below:

```cmake
string(REPLACE "{_IMPORT_PREFIX}/tools/${PORT}" "{_IMPORT_PREFIX}/bin" _contents "${_contents}")
string(REPLACE "{_IMPORT_PREFIX}/tools/debug/${PORT}" "{_IMPORT_PREFIX}/debug/bin" _contents "${_contents}")
```

Comment (yurybura) [reply ports/llvm/portfile.cmake]: @d-winsor, Could you provide your test example? I've tested this port with Hilide only. I think debug version of llvm-config will report incorrect paths for include at least.
Thank  you.

Comment (d-winsor) [reply ports/llvm/portfile.cmake]: That paths for includes/libraries/bin are all correct it just that the content in /debug/bin is being deleted and the paths in `share/clang/**-debug.cmake` files are of the form `{_IMPORT_PREFIX}/tools/debug/${PORT}` not `{_IMPORT_PREFIX}/tools/${PORT}`

I can't share the sample I am testing on but something as simple as should hit the missing file error:
```cpp
cmake_minimum_required(VERSION 3.16)
find_package(LLVM CONFIG REQUIRED)
find_package(Clang CONFIG REQUIRED)
project(test)
add_executable(test test.cpp)
target_link_libraries(test PRIVATE clangAST)
```

Comment (yurybura) [reply ports/llvm/portfile.cmake]: @d-winsor, Thank you for report this. Let me check.

**Comment by JackBoosY** [comment]
Note the other paths in the cmake file.
Related: #9622.

File: ports/llvm/portfile.cmake [hunk 3]
```diff
@@ -1,132 +1,130 @@
...
-)
+vcpkg_find_acquire_program(PYTHON3)
 
 vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
+    SOURCE_PATH ${SOURCE_PATH}/llvm
     PREFER_NINJA
-    OPTIONS ${FEATURE_OPTIONS}
-        -DLLVM_TARGETS_TO_BUILD=X86
-        -DLLVM_ABI_BREAKING_CHECKS=FORCE_OFF
```
Comment (d-winsor) [inline ports/llvm/portfile.cmake]: Why was this removed? Will this not cause a linker errors in debug (as this defaults to WITH_ASSERTS)?

Atleast this is what I see when I try an link c++14 (llvm) libraries with c++17 which should be ABI compatiable.

```
... error LNK2038: mismatch detected for 'LLVM_ENABLE_ABI_BREAKING_CHECKS': value '1' doesn't match value '0' in ...
```

File: ports/llvm/portfile.cmake [hunk 4]
```diff
@@ -1,132 +1,130 @@
...
-        -DLLVM_TOOLS_INSTALL_DIR=tools/llvm
-        -DLLVM_PARALLEL_LINK_JOBS=1
```
Comment (d-winsor) [inline ports/llvm/portfile.cmake]: Even my 16 core 3950x with 32GB of ram restarts the build becuase of memory overuse. Consider leaving this which may also resolve issues with the CI.
```
-- Restarting Build without parallelism because memory exceeded
```

File: ports/llvm/portfile.cmake (hunk 3 above)
Comment (yurybura) [reply ports/llvm/portfile.cmake]: d-winsor, I followed the LLVM's release script: https://github.com/llvm/llvm-project/blob/release/10.x/llvm/utils/release/build_llvm_package.bat

I'll add new feature for control LLVM_ABI_BREAKING_CHECKS...

File: ports/llvm/portfile.cmake (hunk 4 above)
Comment (yurybura) [reply ports/llvm/portfile.cmake]: @d-winsor

> ```
> -- Restarting Build without parallelism because memory exceeded
> ```
As I understand correctly, to avoid this error I should call `vcpkg_install_cmake(DISABLE_PARALLEL)`.

But OK, let me try to disable parallel linking in the LLVM.

File: ports/llvm/portfile.cmake (hunk 2 above)
Comment (yurybura) [reply ports/llvm/portfile.cmake]: @d-winsor, It should works now. Please check.

File: ports/llvm/portfile.cmake (hunk 3 above)
Comment (yurybura) [reply ports/llvm/portfile.cmake]: Resolved.

File: ports/llvm/portfile.cmake (hunk 4 above)
Comment (yurybura) [reply ports/llvm/portfile.cmake]: Resolved. But LINK : fatal error LNK1201 still present on CI.

File: ports/llvm/portfile.cmake (hunk 2 above)
Comment (d-winsor) [reply ports/llvm/portfile.cmake]: I believe there is still a problem here. The format of the paths in the `*-debug.cmake` files is different from `*-release.cmake` files. You should be replacing the following for debug instead:
```cmake
{_IMPORT_PREFIX}/debug/tools/${PORT}
```

Comment (yurybura) [reply ports/llvm/portfile.cmake]: Sorry for this (I can test it only tomorrow). Let me fix it shortly.

Comment (d-winsor) [reply ports/llvm/portfile.cmake]: No problems. I really appreciate you making this contribution, it is really useful to my current work.

**Comment by Voskrese** [comment]
2 days ago · LLVM 10.0.0 Release

vcpkg_check_linkage(ONLY_STATIC_LIBRARY) 

> why only static build? after, other packages will have errors when linking

dynamic & static

https://github.com/vovkos/llvm-package-windows/releases
https://github.com/ziglang/zig/wiki/How-to-build-LLVM,-libclang,-and-liblld-from-source
https://github.com/llvm/llvm-project/releases

**Comment by JackBoosY** [comment]
Any news?

File: ports/llvm/portfile.cmake (hunk 1 above)
Comment (yurybura) [reply ports/llvm/portfile.cmake]: Resolved.

**Comment by yurybura** [comment]
> 2 days ago · LLVM 10.0.0 Release
> 
> vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
> 
> > why only static build? after, other packages will have errors when linking
> 
> dynamic & static
> 
> https://github.com/vovkos/llvm-package-windows/releases
> https://github.com/ziglang/zig/wiki/How-to-build-LLVM,-libclang,-and-liblld-from-source
> https://github.com/llvm/llvm-project/releases

@Voskrese, thank you for your comment.

Shared libraries don't support on Windows:
https://github.com/llvm/llvm-project/blob/master/llvm/CMakeLists.txt#L567

Let's finish this PR for only_static configuration. Then let me try to enable shared libraries (at least for non-windows triplets) in a separate PR.

**Comment by yurybura** [comment]
@JackBoosY I've almost done. Windows CI still doesn't work. Can somebody have have a look?

**Comment by JackBoosY** [comment]
The reason we have these regressions is because the disks of the pipeline machines are full.
So don't worry about that, we will rerun Windows PR test after resolve this problem.

**Comment by dan-shaw** [comment]
@yurybura @JackBoosY Yes, I will be removing `buildtrees/debug-folder` and `buildtrees/rel-folder` after the debug and release installs respectively so there will be more space. Will get this change in during the weekend

File: ports/llvm/portfile.cmake (hunk 2 above)
Comment (yurybura) [reply ports/llvm/portfile.cmake]: I've added `d` postfix to the debug shared libraries and I've copied them to the ./bin directory, Now llvm-config.exe and LLVM CMake scripts should works.
@d-winsor Please let me know if you will have any trouble.

**Comment by yurybura** [comment]
@JackBoosY From my side this PR is ready to merge.
Thank you.

File: ports/llvm/portfile.cmake [hunk 5]
```diff
@@ -1,139 +1,166 @@
...
+        string(REPLACE "{_IMPORT_PREFIX}/debug/bin/" "{_IMPORT_PREFIX}/bin/" _contents "${_contents}")
+        file(WRITE ${_target} "${_contents}")
+    endforeach()
+
+    # Install debug shared libraries in the `/bin` directory
+    file(GLOB _debug_shared_libs ${CURRENT_PACKAGES_DIR}/debug/bin/*${CMAKE_SHARED_LIBRARY_SUFFIX})
+    file(INSTALL ${_debug_shared_libs} DESTINATION ${CURRENT_PACKAGES_DIR}/bin)
+
+    file(REMOVE_RECURSE
+        ${CURRENT_PACKAGES_DIR}/debug/bin
```
Comment (d-winsor) [inline ports/llvm/portfile.cmake:153]: As per our previous discussion you are still deleting the bin folder after correcting to the files above to point to it. Is this a mistake or is it intentional and I am missing something about Vcpkg.

**Comment by yurybura** [comment]
Hello @dan-shaw, @JackBoosY. Please have a look this PR.
Can I temporary add failed configs to the ci.baseline.txt?

**Comment by JackBoosY** [comment]
@yurybura Since llvm is dependent on other ports, I think we can't skip llvm's pipeline test for the time being.

**Comment by JackBoosY** [comment]
llvm test successful on x86-windows.

**Comment by PhoebeHui** [comment]
The failures with x64-windows and x64-windows-static:

oleaut32.lib uuid.lib comdlg32.lib advapi32.lib 
LINK : fatal error LNK1201: error writing to program database 'C:\vsts\_work\4\s\buildtrees\llvm\x64-windows-dbg\bin\llvm-isel-fuzzer.pdb'; check for insufficient disk space, invalid path, or insufficient privilege
ninja: build stopped: subcommand failed.

@dan-shaw, could help have a look again? it seems need more space on CI test machine.

**Review by PhoebeHui — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by piotrek-szczygiel** [comment]
When will I be able to download this version from vcpkg repository?

**Comment by ZeeWanderer** [comment]
COnsider adding all possible LLVM_TARGETS_TO_BUILD as a list of options. Port produces ~80 GB of artifacts on build of llvm[core,utils,tools]:x64-windows while if only X86 backend is built (both Debug and Release) ~28 GB of artifacts are produced. Did not measure compile time but it should be a lot less, thou current list of port options is already quite extensive.

**Comment by yurybura** [comment]
Dear @ZeeWanderer, thank you for the comment. I'm working for support LLVM_TARGETS_TO_BUILD. Unfortunately, I'm not sure that ~28 GB will be acceptable. There are no official opinion from the CI holders.

File: ports/llvm/CONTROL [hunk 6]
```diff
@@ -1,19 +1,75 @@
...
+Description: Build LLVM utils.
 
-Feature: example
-Description: Generate build targets for the LLVM examples
+Feature: target-all
+Description: Build all targets.
 
-Feature: test
-Description: Generate build targets for the LLVM unit tests.
+Feature: target-aarch64
```
Comment (cbezault) [inline ports/llvm/CONTROL:17]: The description should probably be:

Build `architecture` backend.

**Review by cbezault — APPROVED** [review]

File: ports/llvm/CONTROL (hunk 6 above)
Comment (yurybura) [reply ports/llvm/CONTROL:17]: Fixed. Thank you.

**Comment by yurybura** [comment]
Finally CI looks happy)
@strega-nil @PhoebeHui @PhoebeHui Please review/merge this PR.
Thanks!

**Comment by alexreinking** [comment]
@yurybura -- Thank you so much for your work on this! I'm with the Halide project and (among other things) am working on improving our build on Windows. This is a huge help.

**Comment by strega-nil** [comment]
This is extremely exciting! I'm hopeful we can merge today :3

**Comment by tjysdsg** [comment]
Is this PR ready to be merged? I have a project that needs LLVM-10, and it would be great to vcpkg to handle this dependency! Thanks for all your hard work btw

**Comment by alexreinking** [comment]
> Is this PR ready to be merged? I have a project that needs LLVM-10, and it would be great to vcpkg to handle this dependency! Thanks for all your hard work btw

Ditto! Happy to lend a hand any way I can to grease the skids here, too 🙂

**Comment by strega-nil** [comment]
Ah, sorry! I got distracted and forgot to hit the button... unfortunately I'm still learning the ropes -.-

**Comment by alexreinking** [comment]
> Ah, sorry! I got distracted and forgot to hit the button... unfortunately I'm still learning the ropes -.-

No, no, thank you so much! This is a big help to me and my team!

**Comment by cenit** [comment]
sadly, llvm is broken on Linux now.

**Comment by strega-nil** [comment]
@cenit looks like running out of space, so I'm gonna rerun it. This is the issue with LLVM

**Comment by cenit** [comment]
It’s broken  also on my local machine now. Still not investigated the reason, but for sure not disk space

**Comment by strega-nil** [comment]
That's really odd... I'm looking into it

**Comment by yurybura** [comment]
My local build works fine. Debug build log is attached:
[install-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/4577243/install-x64-linux-dbg-out.log)

@cenit Can you attach your log?

**Comment by cenit** [comment]
Sorry I lost logs in the update (last step is cleanup of buildtrees folder in any case). In case the same problem happens again, tonight, I will let you know. Seeing that it was broken also on master made maybe too quick to think it was this PR fault. Maybe it was just a double coincidence (two different unrelated problems here and on my machine)

**Comment by cenit** [comment]
re-launched manually to have logs sooner:

```
/usr/bin/ar: creating t.a
CMake Error at lib/WindowsManifest/CMakeLists.txt:11 (get_filename_component):
  get_filename_component unknown component optimized


-- Version: 0.0.0
-- Performing Test HAVE_GNU_POSIX_REGEX
-- Performing Test HAVE_GNU_POSIX_REGEX
-- Performing Test HAVE_GNU_POSIX_REGEX -- failed to compile
-- Performing Test HAVE_POSIX_REGEX
-- Performing Test HAVE_POSIX_REGEX
-- Performing Test HAVE_POSIX_REGEX -- success
-- Performing Test HAVE_STEADY_CLOCK
-- Performing Test HAVE_STEADY_CLOCK
-- Performing Test HAVE_STEADY_CLOCK -- success
```

Distributor ID: Ubuntu
Description:    Ubuntu 20.04 LTS
Release:        20.04
Codename:       focal

**Comment by strega-nil** [comment]
It's working on my machine... how do you run the build?

**Comment by BillyONeal** [comment]
Under the hypothesis that it failed in CI due to the OOM killer, this makes the test VMs have more RAM: https://github.com/microsoft/vcpkg/pull/11174 (The PR infrastructure under which this PR passed had half as much RAM per core)

**Comment by strega-nil** [comment]
The LLVM in the new infra is now failing in the same way as in @cenit's case. This is really, really weird.

**Comment by BillyONeal** [comment]
Considering it turns out not to be the OOM killer and we have direct customer feedback of it being broken on their machine, should we back this out?

**Comment by strega-nil** [comment]
Yeah, I guess so. I don't like having such an old version, but we can at least try and figure out why it's broken.

**Comment by cenit** [comment]
For me it's ok to just wait a little bit and have llvm 10 properly installed. It just depends on timing, maybe @yurybura can help?

**Comment by yurybura** [comment]
@cenit Your issue is related to the external dependency to the libxml2. It easy to fix, but I need full CMake log. This is CMake configuration issue.
@BillyONeal @strega-nil CI linker fail is not related to the issue mentioned above.

**Comment by yurybura** [comment]
@cenit @BillyONeal @strega-nil I found the full log in the PR #11174. Let me try to fix it.

**Comment by cenit** [comment]
thank you. I'd really like to help you but I am really very constrained with time... sorry

**Comment by yurybura** [comment]
@cenit @BillyONeal @strega-nil Please review/merge my fix in the PR #11108.
Thanks a lot.
